unit/java_bytecode/java_bytecode_convert_method/convert_invoke_dynamic.cpp

```diff
@@ -0,0 +1,674 @@
+  REQUIRE_THAT(
+    parents,
+    Catch::Matchers::Vector::ContainsElementMatcher<irep_idt>{
+      test_data.lambda_interface});
+
+  const auto &interface_children =
+    class_hierarchy.get_children_trans(test_data.lambda_interface);
+
+  REQUIRE_THAT(
+    interface_children,
+    Catch::Matchers::Vector::ContainsElementMatcher<irep_idt>{
+      tmp_class_identifier});
+
+  const java_class_typet::componentt super_class_component =
+    require_type::require_component(tmp_lambda_class_type, "@java.lang.Object");
+
+  const symbol_typet &super_class_type = require_type::require_symbol(
+    super_class_component.type(), "java::java.lang.Object");
+
+  const symbolt &base_class_symbol = require_symbol::require_symbol_exists(
+    symbol_table, super_class_type.get_identifier());
+
+  REQUIRE(base_class_symbol.is_type);
+  const class_typet &super_class_type_struct =
+    require_type::require_incomplete_class(base_class_symbol.type);
+
+  require_type::require_component(super_class_type_struct, "@class_identifier");
+  // TODO verify the components of the class have been set correctly
+
+  THEN("The function in the class should call the lambda method")
+  {
+    const irep_idt method_identifier =
+      id2string(tmp_class_identifier) +
+      test_data.lambda_interface_method_descriptor;
+    const symbolt &method_symbol =
+      require_symbol::require_symbol_exists(symbol_table, method_identifier);
+
+    REQUIRE(method_symbol.is_function());
+
+    const std::vector<codet> &assignments =
+      require_goto_statements::get_all_statements(method_symbol.value);
+
+    std::vector<code_function_callt> function_calls =
+      require_goto_statements::find_function_calls(
+        assignments, test_data.lambda_function_id);
+
+    INFO("Looking for function call of " << test_data.lambda_function_id);
+    REQUIRE(function_calls.size() == 1);
+    const code_function_callt &function_call = function_calls[0];
+
+    std::string variable_prefix = id2string(method_identifier) + "::";
+    // replace all symbol exprs with a prefixed symbol expr
+    std::vector<exprt> expected_args = test_data.expected_params;
+    for(exprt &arg : expected_args)
+    {
+      for(auto it = arg.depth_begin(); it != arg.depth_end(); ++it)
+      {
+        if(it->id() == ID_symbol)
+        {
+          symbol_exprt &symbol_expr = to_symbol_expr(it.mutate());
+          const irep_idt simple_id = symbol_expr.get_identifier();
+          symbol_expr.set_identifier(variable_prefix + id2string(simple_id));
+        }
+      }
+    }
+
+    REQUIRE_THAT(
+      function_call.arguments(),
+      Catch::Matchers::Vector::EqualsMatcher<exprt>{expected_args});
+
+    if(test_data.should_return_value)
+    {
+      require_expr::require_symbol(function_call.lhs());
+    }
+    else
+    {
+      REQUIRE(function_call.lhs().is_nil());
+    }
+  }
+}
+
+/// Find the assignment to the lambda and then call validate_lamdba_assignement
+/// for full validation.
```

Comment:
Missing doxygen for arguments.

Reply:
This was me being lazy as I wasn't sure it would help but have taken the comments from the `validate_lambda_assignment` and document the one that's not covered